packages/ember/addon/instance-initializers/sentry-performance.ts

```diff
@@ -73,38 +86,216 @@ export function _instrumentEmberRouter(
     });
   });
 
-  routerService.on('routeDidChange', (transition: any) => {
-    const { toRoute } = getTransitionInformation(transition, routerService);
-    let renderSpan: any;
+  routerService.on('routeDidChange', () => {
     if (!transitionSpan || !activeTransaction) {
       return;
     }
     transitionSpan.finish();
 
-    if (disablePostTransitionRender) {
+    if (disableRunloopPerformance) {
       activeTransaction.finish();
+      return;
+    }
+
+    run.backburner.on('end', finishActiveTransaction);
+  });
+
+  return {
+    startTransaction,
+  };
+}
+
+function _instrumentEmberRunloop(config: typeof environmentConfig['@sentry/ember']) {
+  const { disableRunloopPerformance, minimumRunloopQueueDuration } = config;
+  if (disableRunloopPerformance) {
+    return;
+  }
+
+  let currentQueueStart: number | undefined;
+  let currentQueueSpan: Span | undefined;
+  const instrumentedEmberQueues = [
+    'actions',
+    'routerTransitions',
+    'render',
+    'afterRender',
+    'destroy',
+  ] as EmberRunQueues[];
+
+  run.backburner.on('begin', (_: any, previousInstance: any) => {
+    if (previousInstance) {
+      return;
+    }
+    const activeTransaction = getActiveTransaction();
+    if (!activeTransaction) {
+      return;
+    }
+    if (currentQueueSpan) {
+      currentQueueSpan.finish();
     }
+    currentQueueStart = timestampWithMs();
+
+    instrumentedEmberQueues.forEach(queue => {
+      run.scheduleOnce(queue, null, () => {
```

Comment:
Is this scheduled twice on purpose?

Reply:
Yes, once to put it into the queue and then once in the queue to put it towards the end once the queue is being processed (since they get created on the runloop's begin).